core/src/main/java/io/questdb/cutlass/pgwire/modern/PGPipelineEntry.java

```diff
@@ -2162,36 +2165,76 @@ private void outRecord(PGResponseSink utf8Sink, Record record, int columnCount)
     }
 
     private void outRowDescription(PGResponseSink utf8Sink) {
-        //todo: wide table metadata can overflow the send buffer and corrupt communication
         final RecordMetadata metadata = factory.getMetadata();
-        utf8Sink.put(MESSAGE_TYPE_ROW_DESCRIPTION);
-        final long addr = utf8Sink.skipInt();
         final int n = pgResultSetColumnTypes.size() / 2;
-        utf8Sink.putNetworkShort((short) n);
-        for (int i = 0; i < n; i++) {
-            final int typeFlag = pgResultSetColumnTypes.getQuick(2 * i);
-            final int columnType = toColumnType(ColumnType.isNull(typeFlag) ? ColumnType.STRING : typeFlag);
-            utf8Sink.putZ(metadata.getColumnName(i));
-            utf8Sink.putIntDirect(0); //tableOid ?
-            utf8Sink.putNetworkShort((short) (i + 1)); //column number, starting from 1
-            utf8Sink.putNetworkInt(PGOids.getTypeOid(columnType)); // type
-            if (ColumnType.tagOf(columnType) < ColumnType.STRING) {
-                // type size
-                // todo: cache small endian type sizes and do not check if type is valid - its coming from metadata, must be always valid
-                utf8Sink.putNetworkShort((short) ColumnType.sizeOf(columnType));
-            } else {
-                // type size
-                utf8Sink.putNetworkShort((short) -1);
-            }
+        long messageLengthAddress = 0;
+        if (outResendColumnIndex == 0 && outResendRecordHeader) {
+            utf8Sink.put(MESSAGE_TYPE_ROW_DESCRIPTION);
+            messageLengthAddress = utf8Sink.skipInt();
+            utf8Sink.putNetworkShort((short) n);
+            utf8Sink.bookmark();
+        }
+        final boolean isMsgLengthRequired = messageLengthAddress > 0;
+        try {
+            while (outResendColumnIndex < n) {
+                final int i = outResendColumnIndex;
+                final int typeFlag = pgResultSetColumnTypes.getQuick(2 * i);
+                final int columnType = toColumnType(ColumnType.isNull(typeFlag) ? ColumnType.STRING : typeFlag);
+                utf8Sink.putZ(metadata.getColumnName(i));
+                utf8Sink.putIntDirect(0); //tableOid ?
+                utf8Sink.putNetworkShort((short) (i + 1)); //column number, starting from 1
+                utf8Sink.putNetworkInt(PGOids.getTypeOid(columnType)); // type
+                if (ColumnType.tagOf(columnType) < ColumnType.STRING) {
+                    // type size
+                    // todo: cache small endian type sizes and do not check if type is valid - its coming from metadata, must be always valid
```

Comment:
`ColumnType#sizeOf()` is fast as it uses a lookup table. Considering that, do we really need this todo?